compiler/compiler.go

```diff
@@ -654,10 +654,14 @@ func encode(i uint16) []byte {
 	return b
 }
 
-func kind(node ast.Node) reflect.Kind {
+// return kind and custom_type flag
+func kind(node ast.Node) (reflect.Kind, bool) {
 	t := node.Type()
 	if t == nil {
-		return reflect.Invalid
+		return reflect.Invalid, false
 	}
-	return t.Kind()
+
+	k := t.Kind()
+	name := t.Name()
+	return k, name != "" && name != k.String()
```

Comment:
Is it the right way of checking if it’s custom type?

Reply:
not really,
for string name = 'string' (same for other basic types: int='int', int8='int8', etc...),
name is empty for maps, slices, arrays, functions, channels,
but for anonymous structs name also empty.

another option check `t.PkgPath() != ""`, we will get similar behavior.

Reply:
Maybe we also need to check for anonymous structs than? Or google some canonical way of doing it)

Reply:
I googled and found nothing :( 
I will assume that the best option is type assertion. But we only have `reflect.Type`:
`_, ok := reflect.Zero(t).Interface().(string)` - looks poor

Anyway, I inverted logic. Now flag detects basic types. It works for all basic types without exception.